docs/feature_layers.md

```diff
@@ -121,6 +121,58 @@ layer_state_t layer_state_set_user(layer_state_t state) {
 }
 ```
 
+### Example: Keycode to cycle through layers
+
+This example shows how to implement a custom keycode to cycle through a range of layers.
+
+```c
+// Define the keycode, `QK_USER` avoids collisions with existing keycodes
+enum keycodes {
+  KC_CYCLE_LAYERS = QK_USER,
+};
+
+// 1st layer on the cycle
+#define LAYER_CYCLE_START 0
+// Last layer on the cycle
+#define LAYER_CYCLE_END   4
+
+// Add the behaviour of this new keycode
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case KC_CYCLE_LAYERS:
+      // Our logic will happen on presses, nothing is done on releases
+      if (!record->event.pressed) { 
+        // We've already handled the keycode (doing nothing), let QMK know so no further code is run unnecessarily
+        return false;
+      }
+
+      uint8_t current_layer = get_highest_layer(layer_state);
+
+      // Check if we are within the range, if not quit
+      if (curent_layer > LAYER_CYCLE_END || current_layer < LAYER_CYCLE_START) {
+        return false;
+      }
+
+      // If we are within the range, but not on the last layer, turn next layer ON
+      if (current_layer < LAYER_CYCLE_END) {
+          layer_on(current_layer + 1);
```

Comment:
layer_move might make more sense here.  Cycling, to me, implies that only one of the layers is on, and that we're cycling through which layer is on.  So, if there are transparencies in these layers, it may cause some confusion

Also, this way, you could use this for this block and the next: 
```c
    current_layer++;
    if (current_layer > LAYER_CYCLE_END) {
        current_layer = LAYER_CYCLE_START;
    }
    layer_move(current_layer);
```